geopandas/geodataframe.py

```diff
@@ -865,7 +865,7 @@ def iterfeatures(self, na="null", show_bbox=False, drop_id=False):
         if not self.columns.is_unique:
             raise ValueError("GeoDataFrame cannot contain duplicated column names.")
 
-        properties_cols = self.columns.difference([self._geometry_column_name])
+        properties_cols = [c for c in self.columns if c != self._geometry_column_name]
```

Comment:
```suggestion
        properties_cols = self.columns.drop(self.geometry.name)
```

I would do this to keep `properties_cols` an `pandas.Index`. It makes very little difference but is seems to be a bit cleaner.